---
title: "Intro to Transcriptomics"
author: Nicole Black, Wade Boohar, Kayla Xu
date: 07/17/22
updated: 09/26/23
---

***Deliverables***
-Upload this R Notebook to your GitHub and submit the link to your Repo on BB.
-Include any graphs or figures created in this assignment in the folder with your R notebook with descriptive file names.

Since this is an optional partner activity, it is okay if your answers are the same as your partnerâ€™s as long as everyone understands it and could explain it in their own words if asked. Each person must individually push their code to Github. *At the top of your R Notebook, write the name of you and your partner(s) as a comment.*

***Complete the following coding activity and answer any following questions as comments in your R Notebook***

In Transcriptomics_Tutorial.Rmd, you learned how to manipulate the SummarizedExperiment data structure, turn it into more readable dataframes, complete basic analysis using those dataframes, and performed a differential expression analysis based on age. Before saving the rna_counts, rna_clinical, and rna_genes however, we masked out any sample that was not a tumor. In this assignment, you will need to reinstantiate the rna_se object and then perform differential expression analysis based on tumor status.

*Pre-Assignment*
Use knitr function to set your working directory to your analysis_data folder in qbio_490_name.
```{r}
normalizePath("/Users/hunter/Desktop/QBIO_490_HunterN/analysis_data") 
```

Load all necessary packages and instantiate the rna_se object
```{r}
library(TCGAbiolinks)
library(SummarizedExperiment)
```


*1*
Perform all data cleaning and label manipulation steps (hint: applying colnames and rownames to make the dataframes more readable) except for masking out the patient with an NA for age and the "Solid Tissue Normal" sample types.
```{r}
query <- GDCquery(project = "TCGA-BRCA", data.category = "Transcriptome Profiling", data.type = "Gene Expression Quantification")
GDCdownload(query)
se <- GDCprepare(query)
se <- se[, !is.na(colData(se)$age)]
se <- se[, colData(se)$sample_type != "Solid Tissue Normal"]
colnames(se)[1:5] <- c("SampleID", "GeneSymbol", "TranscriptID", "Expression_R1", "Expression_R2")

```


*2*
Determine and define your experimental design for the differential expression analysis. Control for covariance of age, gender, OR vital_status. Manipulate those columns so that they are ready for differential expression analysis (hint: what kind of variables are they? what data type are they by default? do you need to handle unknown values?)
```{r}
library(DESeq2)

counts = SummarizedExperiment(DFrame())
metadata = SummarizedExperiment(GRanges())

dds <- DESeqDataSetFromMatrix(countData = counts,
                              colData = metadata,
                              design = ~ age + gender + vital_status)

dds$age <- as.numeric(dds$age)
dds$vital_status <- as.factor(dds$vital_status)
dds$gender <- as.factor(dds$gender)
dds <- DESeq(dds)
results <- results(dds)
topGenes <- head(results[order(results$padj), ], n = 10)
print(topGenes)

```


*3*
Perform the differential expression analysis, create a volcano plot, and interpret what genes from each of the 6 sections means (assuming thresholds of -log10padj = 2 and log2FoldChange = 1)
```{r}
dds <- DESeq(dds)
results <- results(dds)
threshold_padj <- 2
threshold_log2FoldChange <- 1
significant_genes <- subset(results, padj < 10^(-threshold_padj) & abs(log2FoldChange) > threshold_log2FoldChange)

# Load necessary libraries for plotting
library(ggplot2)

volcano_plot <- ggplot(results, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(color = ifelse(padj < 10^(-threshold_padj) & abs(log2FoldChange) > threshold_log2FoldChange, "Significant", "Not Significant"))) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "red") + 
  geom_vline(xintercept = c(-threshold_log2FoldChange, threshold_log2FoldChange), linetype = "dashed", color = "blue") +
  labs(x = "log2(Fold Change)", y = "-log10(padj)") +
  theme_minimal() +
  ggtitle("Volcano Plot")

print(volcano_plot)

```
Cannot get DESeq to download so volcano cannot print. However, I can predict what will plot in each section.

top-right genes: low p-value, most upregulated, high log2fold change
bottom-right genes: high negative log2fold change, downregulated, low p-value
top-left genes: not significant p-values, high log2fold
bottom-left genes: high negative log2fold, not significant p-values
top-middle genes: slightly high log2fold, slightly higher p-values than bottom-middle
bottom-middle genes: lower p-values, lower log2fold

Save the picture of the volcano plot (using either ggsave() or right clicking and manually downloading the image and push this .Rmd and the image to GitHub)
